### ISSUE TITLE
Expose IndexCollection in Python

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#77]
Multiple people have asked for the ability to perform indexing from Python. This shouldn't be too hard, we just need to properly expose [`IndexCollection`](https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/index/IndexCollection.java).

**Comment by lintool** [comment]
For those needing something immediate, use a `JsonCollection`:
https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/collection/JsonCollection.java#L36

That is, write a simple script to reformat your collection into JSON, as specified above. Then you can use the MS MACRO instructions: https://github.com/castorini/anserini/blob/master/docs/experiments-msmarco-passage.md

**Comment by lintool** [comment]
@yuki617 might be interested in taking this one.

tl;dr - make this work in Python:

```
sh ./target/appassembler/bin/IndexCollection -collection JsonCollection \
 -generator DefaultLuceneDocumentGenerator -threads 9 -input collections/msmarco-passage/collection_jsonl \
 -index indexes/msmarco-passage/lucene-index-msmarco -storePositions -storeDocvectors -storeRaw 
```

**Comment by lintool** [comment]
Closed by #195
